src/lib/cli/keyfctl.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+const
+  Promise     = require('bluebird'),
+  _           = require("lodash"),
+  debug       = require('debug')('keyfctl'),
+  capitano    = require('capitano'),
+  core        = require('../shared/core'),
+  utils       = require('../shared/utils')
+
+const help = (params, options) => {
+  console.log(`Usage: keyfctl [COMMANDS] [OPTIONS]`)
+  console.log('\nCommands:\n')
+
+  for (let command of capitano.state.commands) {
+    if (command.isWildcard()) continue
+    console.log(`\t${command.signature}\t\t\t${command.description}`)
+  }
+}
+
+const parseGlobalOpts = (options) => {
+  if (options.help) {
+    help()
+    process.exit(0)
+  }
+}
+
+capitano.command({
+  signature: 'help',
+  description: 'output general help page',
+  action: help
+})
+
+capitano.globalOption({
+  signature: 'help',
+  boolean: true,
+  alias: ['h'],
+})
+
+capitano.globalOption({
+  signature: 'verbose',
+  boolean: true,
+  alias: [ 'v' ],
+  required: false
+})
+
+capitano.globalOption({
+  signature: 'vv', // this is a hack until we have multiple bool support
+  boolean: true,
+  required: false
+})
+
+capitano.command({
+  signature: 'k8s',
+  options: [{
+    signature: 'dry-run',
+    boolean: true,
+    alias: [ 'n' ],
+    required: false
+  }],
+  action: (params, options) => {
+    parseGlobalOpts(options)
+
+    if (options.vv) {
+      Promise.longStackTraces()
+      utils.inspect('options: ', options)
+    }
+
+    let res = core.generateFrames('HEAD')
+    .catch(err => {
+      throw new Error(err)
+    })
+
+    if (options.verbose) {
+      res.tap(frames => {
+        console.log('==> All frames')
+        for (let frame of frames) {
+          // Output some feedback about the frame history
+          console.log(utils.printFormatFrame(frame))
+        }
+      })
+    }
+
+    res = res.filter(frame => {
```

Comment:
I see what you're doing here with augmenting the promise, but I've never seen it written in this kind of style before. I can't decide if it's disgusting or utterly genius. :)